sunpy/coordinates/frames.py

```diff
@@ -587,6 +587,15 @@ def make_3d(self):
 
             d = np.fmin(d, dd) if self._spherical_screen['only_off_disk'] else dd
 
+        # If we have all NaNs in d, warn user
+        if np.all(np.isnan(d)):
```

Comment:
```suggestion
        if np.all(np.isnan(d)) and np.any(np.isfinite(cos_alpha)):
```
This added check makes sure that we are emitting the warning only when the input is finite, and not when the user has provided NaNs in the first place.